portal-impl/src/com/liferay/portal/service/impl/UserLocalServiceImpl.java

```diff
@@ -6333,6 +6333,7 @@ protected void notifyUser(User user, ServiceContext serviceContext)
 				serviceContext);
 
 			passwordResetURL = StringBundler.concat(
+				"&languageId=", user.getLanguageId(),
```

Comment:
this is wrong because In this way you are creating the URL like ```&languageId=en_UShttp://localhost:8080/****```

You need to put it as the first URL parameter in order to create something like ```http://localhost:8080/***?languageId=en_US&ticketId=**&ticketKey=***```

Reply:
@smottal 
I fixed it now, thanks for alerting me about this

Reply:
Don't work too
``
			passwordResetURL = StringBundler.concat(
				serviceContext.getPortalURL(), serviceContext.getPathMain(),
				updatePasswordURL, "&languageId=", user.getLanguageId(),
				"&ticketId=", ticket.getTicketId(), "&ticketKey=",
				ticket.getKey());

```